src/Sparrow/Json/LazyStringValue.cs

```diff
@@ -176,17 +188,34 @@ public int CompareTo(object obj)
             throw new NotSupportedException($"Cannot compare LazyStringValue to object of type {obj.GetType().Name}");
         }
 
+        public bool IsDisposed => AllocatedMemoryData == null && _buffer == null;        
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
The whole _point_ of moving the throwing to a different method is to reduce the cost of the throw if it is mostly not done.
You are moving this to a separate method, but force inlining, then you are putting the IsDispose check right here.
You need to move the if outside, not have inlining and only have the throw here.